wrappers/services.go

```diff
@@ -191,6 +191,7 @@ Requires=snapd.frameworks.target{{ if .App.Socket }} {{.SocketFileName}}{{end}}
 X-Snappy=yes
 
 [Service]
+ExecPreStart=mkdir -p {{.App.Snap.RootUserDataDir}} {{.App.Snap.RootCommonUserDataDir}}
```

Comment:
Landing this today won't mean it'll be fixed today either. Can we do the real thing instead?

Note that this particular fix is only helping on edge cases as well, in which a service tries to use the USER_DATA directory, which is arguably a bit awkward on itself.